src/modules/task.js

```diff
@@ -1,46 +1,144 @@
+    listArr.push({
+      id: listArr.length,
+      name: todoTask,
+      done: false,
+    });
+    addTodo(todoTask, listArr.length - 1, false);
+    localStorage.setItem('Todo', JSON.stringify(listArr));
   }
+  taskInput.value = '';
 };
 
-const enterTask = (event) => {
-  error.style.display = 'none';
-  if (event.keyCode === 13 && taskInput.value !== '') {
-    buildTask();
-    taskInput.value = '';
+taskInput.addEventListener('keypress', (event) => {
+  if (event.keyCode === 13) {
+    addTask();
   }
+});
+
+const completeTask = (element) => {
+  element.classList.toggle(check);
+  element.classList.toggle(uncheck);
+  element.parentNode.querySelector('.text-todo').classList.toggle(lineThrough);
+
+  listArr[element.id].done = !listArr[element.id].done;
+};
+
+const updateIndex = () => {
+  listArr.forEach((element, index) => {
+    element.id = index;
+  });
 };
 
-addBtn.addEventListener('click', addTask);
-taskInput.addEventListener('keypress', enterTask);
+const removeTask = (element) => {
+  listArr.splice(element.id, 1);
+  updateIndex();
+  taskList.innerHTML = '';
+  loadTodo(listArr);
+};
+
+const updateTask = (id, value) => {
+  listArr[id].name = value;
+};
+
+taskList.addEventListener('click', (event) => {
+  const element = event.target;
+  const elementJob = element.attributes.job.value;
+  if (elementJob === 'complete') {
+    completeTask(element);
+  } else if (elementJob === 'delete') {
+    removeTask(element);
+  } else if (elementJob === 'edit') {
+    element.parentNode.classList.add('disable');
+    element.parentNode.nextElementSibling.classList.remove('disable');
+    const description = [
+      ...element.parentNode.parentNode.previousElementSibling.children,
+    ];
+    description.forEach((element) => {
+      if (element.id === 'description') {
+        element.classList.add('disable');
+      } else if (element.id === 'update-desc') {
+        element.classList.remove('disable');
+        element.value = element.previousElementSibling.textContent;
+      }
+    });
+  } else if (elementJob === 'save') {
+    element.parentNode.classList.add('disable');
+    element.parentNode.previousElementSibling.classList.remove('disable');
+    const description = [
+      ...element.parentNode.parentNode.previousElementSibling.children,
+    ];
+    let value = '';
+    let span = '';
+    description.forEach((element) => {
+      if (element.id === 'description') {
+        element.classList.remove('disable');
+        span = element;
+      } else if (element.id === 'update-desc') {
+        value = element.value;
+        span.textContent = value;
+        element.classList.add('disable');
+      }
+    });
+    updateTask(element.parentNode.nextElementSibling.children[0].id, value);
+  }
+  localStorage.setItem('Todo', JSON.stringify(listArr));
+});
+
+addBtn.addEventListener('click', () => {
+  addTask();
+});
+
+// resetBtn.addEventListener("click", () => {
+//   localStorage.clear();
+// });
```

Comment:
- Kindly make sure to remove this commented out lines of code to too, make your code cleaner :recycle: